test/test_routes.py

```diff
@@ -254,6 +254,31 @@ def test_get_allowed_paths(self):
         assert len(file_response.text) == len(media_data.BASE64_IMAGE)
         io.close()
 
+    def test_response_attachment_format(self):
+        image_file = tempfile.NamedTemporaryFile(mode="w", delete=False, suffix=".png")
+        image_file.write(media_data.BASE64_IMAGE)
+        image_file.flush()
+
+        html_file = tempfile.NamedTemporaryFile(mode="w", delete=False, suffix=".html")
+        html_file.write("<html>Hello, world!</html>")
+        html_file.flush()
+
+        io = gr.Interface(lambda s: s.name, gr.File(), gr.File())
+        app, _, _ = io.launch(
+            prevent_thread_lock=True,
+            allowed_paths=[
+                os.path.dirname(image_file.name),
+                os.path.dirname(html_file.name),
+            ],
+        )
+        client = TestClient(app)
+
+        file_response = client.get(f"/file={image_file.name}")
+        assert file_response.headers["Content-Type"] == "image/png"
+
+        file_response = client.get(f"/file={html_file.name}")
+        assert file_response.headers["Content-Type"] == "application/octet-stream"
```

Comment:
Can we also test the `content_disposition_type` in these tests?

Reply:
Done. This was giving me issues earlier cause apparently the `content_disposition_type` was being dropped since there was no explicit filename in the response. I also had to set the filename in FileResponse to be the filename of the file requested. It seems like setting the media_type to `octet-stream` was enough to make it download as an attachment previously, but good to have an explicit "attachment" disposition as well.

Reply:
Ah interesting. Nice!